charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -195,7 +195,19 @@ receivers:
 {{- end }}
 
 {{- define "opentelemetry-collector.applyClusterMetricsConfig" -}}
-{{- $config := mustMergeOverwrite (dict "service" (dict "pipelines" (dict "metrics" (dict "receivers" list)))) (include "opentelemetry-collector.clusterMetricsConfig" .Values | fromYaml) .config }}
+{{- $config := .config }}
+
+{{- /* Add leader election extension if enabled */ -}}
```

Comment:
Lets move the logic for injecting the extension into its own helper. It can still be invoked within `opentelemetry-collector.applyClusterMetricsConfig`.